### PR TITLE
Restored e2e tests

### PR DESCRIPTION
**Pull request by ccomb**
with a few additions:
- added a way to display the error in case the tests is in 400 while it should be in 200
- don't write the output file if it's empty
- write the test anyway with empty impacts and score if the status is not 200

### CODE REVIEW

**Review by magopian — COMMENTED** [review]
Good idea to not write the file if it's empty. I'd go one step further: if the status isn't 200, still write to the output file (not sure with which impacts), to make sure we never lose any tests?

File: tests/server.spec.js [hunk 1]
```diff
@@ -544,8 +544,12 @@ afterAll(() => {
   // with their contents.
   function writeE2eResult(key) {
     const target = `${__dirname}/e2e-${key}-output.json`;
-    fs.writeFileSync(target, JSON.stringify(e2eOutput[key], null, 2) + "\n");
-    console.info(`E2e ${key} tests output written to ${target}.`);
+    if (e2eOutput[key].length === 0) {
```
Comment (magopian) [inline tests/server.spec.js:547]: I believe this shouldn't happen anymore, but it's still good to have I guess, to prevent any future accidents ;)

**Review by magopian — APPROVED** [review]
Not sure about the removal of the "market for electricity" from the processes_impacts.json, not sure if it's related to this current PR, and why we don't want it anymore (not used?).

File: tests/server.spec.js [hunk 2]
```diff
@@ -577,6 +581,9 @@ async function expectListResponseContains(path, object) {
 }
 
 function expectStatus(response, code, type = "application/json") {
+  if (response.status === 400 && code != 400) {
```
Comment (n1k0) [inline tests/server.spec.js:584]: Nit: maybe rename the `code` arg to `expectedCode` as the function body is confusing to read

**Review by n1k0 — APPROVED** [review]
LGTM, thanks!

File: tests/server.spec.js (hunk 1 above)
Comment (ccomb) [reply tests/server.spec.js:547]: The PT elec is unused, I'll remove it in another PR

**Pull request merged**